src/scripts/generateBridgeList.ts

```diff
@@ -0,0 +1,160 @@
+  bridgeInfo?: Record<
+    SupportedChainId,
+    {
+      tokenAddress?: string
+    }
+  >
+}
+
+async function generateBridgedList(
+  chainId: SupportedChainId,
+  tokenListSource: string | TokenList,
+  bridgeContractAddress: string,
+  bridgeContractAbi: ContractInterface,
+  methodName: string,
+  outputFilePath: string,
+  tokensToReplace: Record<string, string | null> = {},
+): Promise<void> {
+  const provider = getProvider(chainId, undefined)
+  const bridgeContract = new Contract(bridgeContractAddress, bridgeContractAbi, provider)
+  const multicall = new Contract(MULTICALL_ADDRESS, MULTICALL_ABI, provider)
+
+  // Original source token list
+  const tokensList = (
+    typeof tokenListSource === 'string' ? await fetch(tokenListSource).then((res: any) => res.json()) : tokenListSource
+  ) as TokenList
+
+  const calls = tokensList.tokens.reduce<{ token: TokenInfo; call: Call }[]>((acc, token) => {
+    // Take only Mainnet tokens
+    if (token.chainId !== SupportedChainId.MAINNET) return acc
+    // Do not check for tokens we want to skip/replace
+    if (tokensToReplace[token.address.toLowerCase()] === null) return acc
+
+    // A call to resolve token address in target chain
+    const call = {
+      target: bridgeContract.address,
+      callData: bridgeContract.interface.encodeFunctionData(methodName, [token.address]),
+    }
+
+    acc.push({ token, call })
+
+    return acc
+  }, [])
+
+  const tokens = await multicall.callStatic
+    .tryAggregate(
+      false,
+      calls.map(({ call }) => call),
+    )
+    .then((results) => {
+      return results.reduce((acc: TokenInfo[], res: any, index: number) => {
+        const token = calls[index]?.token
+
+        // Address of the token in the mapped chain
+        const address = '0x' + res[1].slice(-40)
+
+        // Get rid of tokens that are not known by the bridge contract
+        if (!token || address === ZERO_ADDRESS) return acc
+
+        const toReplace = tokensToReplace[token.address.toLowerCase()]
+        if (toReplace !== undefined) {
+          if (toReplace === null) {
+            // Do not add to the list
+            return acc
+          }
+
+          // Replace the fetched address with the one provided
+          acc.push({
+            ...token,
+            chainId,
+            address: toReplace,
+            extensions: undefined,
+          })
+
+          return acc
+        }
+        // Copy everything from the Mainnet token and override address and chainId
+        // Let's hope no one deploys the same token in a different networks with different data
+
+        const extensions = token.extensions as Extensions
+        const tokenAddress = extensions?.bridgeInfo?.[chainId]?.tokenAddress
+
+        if (tokenAddress && tokenAddress.toLowerCase() !== address.toLowerCase()) {
+          console.log(
+            `TokenList extension contains address different than returned by bridge. Extension: ${tokenAddress} Bridge: ${address}`,
+          )
```

Comment:
Uniswap token list includes the address of bridged tokens for some chains.
Use that when available, if different from the bridged address.